streaming/src/main/scala/io/gearpump/streaming/task/Subscription.scala

```diff
@@ -100,7 +100,7 @@ class Subscription(
   def sendMessage(msg: Message, partition: Int): Int = {
 
     // only send message whose timestamp matches the lifeTime
-    if (life.contains(msg.timestamp)) {
+    if (partition >= 0 && life.contains(msg.timestamp)) {
```

Comment:
Is partition expected to have minus value?
Will require(http://www.scala-lang.org/api/2.10.2/index.html#scala.Predef$) better?